components/ntp_background_images/browser/ntp_background_images_utils.cc

```diff
@@ -5,13 +5,39 @@
 
 #include "brave/components/ntp_background_images/browser/ntp_background_images_utils.h"
 
+#include <string>
+
+#include "brave/components/ntp_background_images/browser/url_constants.h"
 #include "brave/components/ntp_background_images/common/pref_names.h"
-#include "components/pref_registry/pref_registry_syncable.h"
+#include "components/prefs/pref_registry_simple.h"
 
 namespace ntp_background_images {
 
-void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
-  registry->RegisterBooleanPref(prefs::kNewTabPageShowBackgroundImage, true);
+void RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
+  registry->RegisterDictionaryPref(
+      prefs::kNewTabPageCachedSuperReferralComponentInfo);
+  registry->RegisterStringPref(
+      prefs::kNewTabPageCachedSuperReferralComponentData, std::string());
+  registry->RegisterStringPref(
+      prefs::kNewTabPageCachedSuperReferralCode, std::string());
+  registry->RegisterListPref(
+      prefs::kNewTabPageCachedSuperReferralFaviconList);
+  registry->RegisterBooleanPref(
+      prefs::kNewTabPageGetInitialSRComponentInProgress, false);
+}
+
+bool IsValidSuperReferralComponentInfo(const base::Value& component_info) {
```

Comment:
this also seems like it would make more sense as a method on the service? Is there a reason to put it in a separate util file?

Reply:
Moved this to service. I added this api here because it's also used in test cases.
But, test cases can use service's private method. So. this also moved to service.